src/openrct2/actions/ParkSetLoanAction.cpp

```diff
@@ -40,19 +40,18 @@ void ParkSetLoanAction::Serialise(DataSerialiser& stream)
 
 GameActions::Result ParkSetLoanAction::Query() const
 {
-    auto currentLoan = gBankLoan;
-    auto loanDifference = currentLoan - _value;
-    if (_value > currentLoan && _value > gMaxBankLoan)
+    auto loanDifference = gBankLoan - _value;
+    if (_value > gBankLoan && _value > gMaxBankLoan)
     {
         return GameActions::Result(
             GameActions::Status::Disallowed, STR_CANT_BORROW_ANY_MORE_MONEY, STR_BANK_REFUSES_TO_INCREASE_LOAN);
     }
-    // FIXME: use money64 literal once it is implemented
-    if (_value < currentLoan && _value < 0)
+    if (_value < gBankLoan && _value < 0.00_GBP)
     {
         return GameActions::Result(GameActions::Status::InvalidParameters, STR_CANT_PAY_BACK_LOAN, STR_LOAN_CANT_BE_NEGATIVE);
     }
-    if (loanDifference > gCash)
+    // When the player is in debt, they need to be able to increase their loan.
+    if (loanDifference > 0 && loanDifference > gCash)
```

Comment:
`loanDifference` isn't very clear on which way it goes. Does a positive value mean that the loan increases? To make this more clear, you can get rid of the variable altogether and just compare the two values it's based on.
```suggestion
    // When the player is in debt, they need to be able to increase their loan.
    if (gBankLoan > _value && gBankLoan - _value > gCash)
```